solver/src/settlement_submission.rs

```diff
@@ -92,8 +98,45 @@ async fn simulate_settlement(
     contract: &GPv2Settlement,
 ) -> Result<()> {
     let method = retry::settle_method_builder(contract, settlement.clone());
-    let data = method.tx.data.as_ref().expect("no data").0.as_slice();
-    tracing::info!("Settlement call: {}", hex::encode(data));
-    method.call().await.context("settle simulation failed")?;
-    Ok(())
+    let tx = method.tx.clone();
+    let result = method.call().await;
+    match &result {
+        Ok(_) => Ok(()),
+        Err(err) => {
+            let context = if is_smart_contract_error(err) {
+                let tenderly_link = tenderly_link(&contract.raw_instance().web3(), tx)
+                    .await
+                    .unwrap_or_else(|err| {
+                        format!("Unable to create simulation link due to: {}", err)
+                    });
+                format!("Settle simulation failed. Link: {}", tenderly_link)
+            } else {
+                "Settle simulation failed.".into()
+            };
+            result.map(|_| ()).context(context)
+        }
+    }
+}
+
+fn is_smart_contract_error(error: &MethodError) -> bool {
+    matches!(error.inner, ExecutionError::Failure(_))
+        || matches!(error.inner, ExecutionError::Revert(_))
+        || matches!(error.inner, ExecutionError::InvalidOpcode)
+}
+
+// Creates a simulation link in the gp-v2 tenderly workspace
+async fn tenderly_link(
+    web3: &Web3<DynTransport>,
+    tx: TransactionBuilder<DynTransport>,
+) -> Result<String> {
+    let current_block = web3.eth().block_number().await?;
+    let network_id = web3.net().version().await?;
+    Ok(format!(
+        "https://dashboard.tenderly.co/gp-v2/staging/simulator/new?block={}&blockIndex=0&from={:#x}&gas=8000000&gasPrice=0&value=0&contractAddress={:#x}&rawFunctionInput=0x{}&network={}",
```

Comment:
Also tiny nit: the org and project name should likely be configurable (in case others want to run solvers for example). If we pass down these values as configuration for the Tenderly link, then we can also do as @vkgnosis suggested and pass the network ID as well (to avoid a failure for creating the link).

Reply:
This would require making it configurable all the way up to the command line, wouldn't it (I guess we can leave default values, yet it seems intrusive, requiring documentation etc. all for such a non core functionality)? 

Maybe we can add that once we have someone who really wants to use this link but is in a different org?

Reply:
Of course. This wasn't a suggesting doing it in this PR. Just something to keep in mind (maybe just add an issue)?

Reply:
I love cleanup!